### PR TITLE
DAOS-3705 test: Get container ACL

### PR DESCRIPTION
**Pull request by amjustin13**
Test covers the following:
SRS-10: Container Management - container ACL allows add/remove access control - https://jira.hpdd.intel.com/browse/SRS-10

- Test that container get-acl command performs as expected with valid inputs in command line for the --verbose and --outfile options.
- Test that container get-acl command fails when attempting to overwrite an existing file.
- Test that container get-acl command fails with no permission -1001 when user doesn't have the right permissions.

Signed-off-by: Justiniano-pagn amanda.justiniano-pagn@intel.com

### CODE REVIEW

File: src/tests/ftest/security/cont_get_acl.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+                acl.append(entry.strip())
+
+        return acl
+
+    @fail_on(CommandFailure)
+    def test_acl_get_valid(self):
+        """
+        JIRA ID: DAOS-3705
+
+        Test Description: Test that container get command performs as
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_get_acl.py:125]:
```suggestion
        Test Description: Test that container get-acl command performs as
```

File: src/tests/ftest/security/cont_get_acl.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+pool:
+  control_method: dmg
+  mode: 511
+  scm_size: 2G
+  name: daos_server
+container:
+  control_method: daos
+  type: POSIX
+
+out_filename:
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_get_acl.yaml:20]: Are these invalid file names? If so you might want to use invalid_out_filename or something. You have vAliD.txt, so I can't tell whether these are invalid names.

File: src/tests/ftest/security/cont_get_acl.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+        If provided  prev_acl file information is different from current acl
+        file information test will fail if flag=True. If flag=False, test will
+        fail in the case that the acl contents are found to have no difference.
+
+        Args:
+            prev_acl (list): list of acl entries within acl-file.
+                Defaults to True.
+            flag (bool): if True, test will fail when acl-file contents are
+                different, else test will fail when acl-file contents are same.
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_get_acl.py:90]:
```suggestion
            flag (bool, optional): if True, test will fail when acl-file contents are
                different, else test will fail when acl-file contents are same. Defaults to True.
```

File: src/tests/ftest/security/cont_get_acl.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+        self.acl_filename = "test_acl_file.txt"
+        self.daos_cmd = self.get_daos_command()
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+        # Get list of ACL entries
+        self.cont_acl = self.get_container_acl_list(
+            self.pool.uuid, self.pool.svc_ranks[0], self.container.uuid)
+
+    def error_handling(self, results, err_msg):
```
Comment (shimizukko) [inline src/tests/ftest/security/cont_get_acl.py:52]: I see this method and acl_file_diff in other test too. You might want to move them to ContSecurityTestBase.

**Review by shimizukko — DISMISSED** [review]
LGTM

File: src/tests/ftest/security/cont_get_acl.py (hunk 1 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_get_acl.py:125]: Will fix.

File: src/tests/ftest/security/cont_get_acl.yaml (hunk 2 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_get_acl.yaml:20]: They are valid filenames, I wanted to make them somewhat abnormal filenames to test the fact that the file should still be created regardless of the name. I'll make the yaml file tag more descriptive.

File: src/tests/ftest/security/cont_get_acl.py (hunk 3 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_get_acl.py:90]: Will fix.

File: src/tests/ftest/security/cont_get_acl.py (hunk 4 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_get_acl.py:52]: Will do.

**Review by shimizukko — APPROVED** [review]
LGTM

File: src/tests/ftest/security/cont_get_acl.py [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        self.daos_cmd.sudo = True
+
+        # Disable raising an exception if the daos command fails
+        self.daos_cmd.exit_status_exception = False
+
+        # Let's check that we can't run as root (or other user) and get
+        # acl information if no permissions are set for that user.
+        test_errs = []
+        for verbose in [True, False]:
+            for outfile in self.params.get("valid_out_filename", "/run/*"):
```
Comment (kjacque) [inline src/tests/ftest/security/cont_get_acl.py:117]: Probably not necessary to do this loop again in this case.

File: src/tests/ftest/security/cont_get_acl.py [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        self.prepare_pool()
+        self.add_container(self.pool)
+
+    @fail_on(CommandFailure)
+    def test_acl_get_valid(self):
+        """
+        JIRA ID: DAOS-3705
+
+        Test Description: Test that container get-acl command performs as
+            expected with invalid inputs.
```
Comment (kjacque) [inline src/tests/ftest/security/cont_get_acl.py:54]: Meant "valid" here?

File: src/tests/ftest/security/cont_get_acl.py (hunk 5 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_get_acl.py:117]: Yeah, it's technically not necessary. The reason I implemented the loop in this case as well was to be thorough in checking that the valid inputs that should pass on the previous case all fail for this case, provided that the user has no permissions. I can definitely change this to check for the permission error in a way that isn't doing the loop again.

File: src/tests/ftest/security/cont_get_acl.py (hunk 6 above)
Comment (amjustin13) [reply src/tests/ftest/security/cont_get_acl.py:54]: Yes, will fix.

**Review by dinghwah — APPROVED** [review]
LGTM.

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by kjacque — APPROVED** [review]

**Comment by amjustin13** [comment]
Test failures CI:
- h5vl_test_parallel failing with: data verification failed for chunk 9: https://jira.hpdd.intel.com/browse/DAOS-5597

**Pull request merged**